### PR TITLE
[FLINK-19660][docs][python] Translate page "table_environment.zh.md" into Chinese

### PR DESCRIPTION
**Pull request by wangxlong**
## What is the purpose of the change

Translate page "table_environment.zh.md" into Chinese

## Brief change log

Translate page "table_environment.zh.md" into Chinese

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no / don't know)
  - The runtime per-record code paths (performance sensitive): (no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no / don't know)
  - The S3 file system connector: (no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wangxlong** [comment]
Hi @dianfu, could you help cc in your free time, thanks.

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 1]
```diff
@@ -22,89 +22,89 @@ specific language governing permissions and limitations
...
 
 * This will be replaced by the TOC
 {:toc}
 
-Create a TableEnvironment
+创建 TableEnvironment
 -------------------------
 
-The recommended way to create a `TableEnvironment` is to create from an `EnvironmentSettings` object:
+创建 `TableEnvironment` 的推荐方式是通过 `EnvironmentSettings` 对象中创建:
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:34]:
```suggestion
创建 `TableEnvironment` 的推荐方式是通过 `EnvironmentSettings` 对象创建:
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 2]
```diff
@@ -113,105 +113,105 @@ These APIs are used to create/remove Table API/SQL Tables and write queries:
...
+        <a href="{{ site.pythondocs_baseurl }}/api/python/pyflink.table.html#pyflink.table.TableEnvironment.from_path">链接</a>
       </td>
     </tr>
     <tr>
       <td>
         <strong>sql_query(query)</strong>
       </td>
       <td>
-        Evaluates a SQL query and retrieves the result as a `Table` object. 
+        计算一条 SQL 查询并将查询的结果作为一个 `Table` 对象。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:149]:
```suggestion
        执行一条 SQL 查询，并将查询的结果作为一个 `Table` 对象。 
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 3]
```diff
@@ -331,35 +331,35 @@ These APIs are used to explain/execute jobs. Note that the API `execute_sql` can
...
     </tr>
     <tr>
       <td>
         <strong>create_statement_set()</strong>
       </td>
       <td>
-        Creates a StatementSet instance which accepts DML statements or Tables.
-        It can be used to execute a multi-sink job. 
+        创建一个可接受 DML 语句或表的 StatementSet 实例。
+        它可用于执行包含多个 sinks 的作业。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:346]:
```suggestion
        它可用于执行包含多个 sink 的作业。
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 4]
```diff
@@ -368,45 +368,42 @@ These APIs are used to explain/execute jobs. Note that the API `execute_sql` can
         <strong>explain(table=None, extended=False)</strong>
       </td>
       <td>
-        Returns the AST of the specified Table API and SQL queries and the execution plan to compute
-        the result of the given `Table` object or multi-sinks plan.
-        If you use the "insert_into" or "sql_update" method to emit data to multiple sinks, you can use this
-        method to get the plan.
-        It can be replaced by <strong>TableEnvironment.explain_sql</strong>, <strong>Table.explain</strong> or <strong>StatementSet.explain</strong>.
+        返回指定 Table API 和 SQL 查询的抽象语法树，以及用来计算给定 `Table` 对象或者多个 sinks 计划结果的执行计划。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:371]:
```suggestion
        返回指定 Table API 和 SQL 查询的抽象语法树，以及用来计算给定 `Table` 对象或者多个 sink 计划结果的执行计划。
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 5]
```diff
@@ -415,103 +412,102 @@ For more details about the different kinds of UDFs, please refer to [User Define
         <strong>create_temporary_function(path, function)</strong>
       </td>
       <td>
-        Registers a Python user defined function class as a temporary catalog function.
+        将一个 Python 用户自定义函数类注册为临时 catalog 函数。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:415]:
```suggestion
        将一个 Python 用户自定义函数注册为临时 catalog 函数。
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 6]
```diff
@@ -415,103 +412,102 @@ For more details about the different kinds of UDFs, please refer to [User Define
...
     </tr>
     <tr>
       <td>
         <strong>create_temporary_system_function(name, function)</strong>
       </td>
       <td>
-        Registers a Python user defined function class as a temporary system function.
-        If the name of a temporary system function is the same as a temporary catalog function,
-        the temporary system function takes precedence.
+        将一个 Python 用户自定义函数类注册为临时系统函数。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:426]:
```suggestion
        将一个 Python 用户自定义函数注册为临时系统函数。
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 7]
```diff
@@ -415,103 +412,102 @@ For more details about the different kinds of UDFs, please refer to [User Define
...
       </td>
     </tr>
     <tr>
       <td>
         <strong>create_java_function(path, function_class_name, ignore_if_exists=None)</strong>
       </td>
       <td>
-        Registers a Java user defined function class as a catalog function under the given path.
-        If the catalog is persistent, the registered catalog function can be used across multiple Flink sessions and clusters.
+        将 Java 用户定义的函数类注册为指定路径下的 catalog 函数。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:438]:
```suggestion
        将 Java 用户自定义函数注册为指定路径下的 catalog 函数。
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 8]
```diff
@@ -415,103 +412,102 @@ For more details about the different kinds of UDFs, please refer to [User Define
...
+        <a href="{{ site.pythondocs_baseurl }}/api/python/pyflink.table.html#pyflink.table.TableEnvironment.create_java_function">链接</a>
       </td>
     </tr>
     <tr>
       <td>
         <strong>create_java_temporary_function(path, function_class_name)</strong>
       </td>
       <td>
-        Registers a Java user defined function class as a temporary catalog function.
+        将 Java 用户定义的函数类注册为临时 catalog 函数。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:450]:
```suggestion
        将 Java 用户自定义函数注册为临时 catalog 函数。
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 9]
```diff
@@ -415,103 +412,102 @@ For more details about the different kinds of UDFs, please refer to [User Define
...
+        <a href="{{ site.pythondocs_baseurl }}/api/python/pyflink.table.html#pyflink.table.TableEnvironment.create_java_temporary_function">链接</a>
       </td>
     </tr>
     <tr>
       <td>
         <strong>create_java_temporary_system_function(name, function_class_name)</strong>
       </td>
       <td>
-        Registers a Java user defined function class as a temporary system function.
+        将 Java 用户定义的函数类注册为临时系统函数。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:461]: ditto

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 10]
```diff
@@ -415,103 +412,102 @@ For more details about the different kinds of UDFs, please refer to [User Define
...
+        <a href="{{ site.pythondocs_baseurl }}/api/python/pyflink.table.html#pyflink.table.TableEnvironment.drop_function">链接</a>
       </td>
     </tr>
     <tr>
       <td>
         <strong>drop_temporary_function(path)</strong>
       </td>
       <td>
-        Drops a temporary system function registered under the given name.
+        删除指定名称下已注册的临时函数。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:483]:
```suggestion
        删除指定名称下已注册的临时系统函数。
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 11]
```diff
@@ -520,41 +516,41 @@ For more details about the different kinds of UDFs, please refer to [User Define
         <strong>register_function(name, function)</strong>
       </td>
       <td>
-        Registers a Python user-defined function under a unique name. 
-        Replaces already existing user-defined function under this name.
-        It can be replaced by <strong>create_temporary_system_function</strong>.
+        注册一个具有唯一名称的Python 自定义函数。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:519]:
```suggestion
        注册一个 Python 用户自定义函数，并为其指定一个唯一的名称。 
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 12]
```diff
@@ -520,41 +516,41 @@ For more details about the different kinds of UDFs, please refer to [User Define
         <strong>register_function(name, function)</strong>
       </td>
       <td>
-        Registers a Python user-defined function under a unique name. 
-        Replaces already existing user-defined function under this name.
-        It can be replaced by <strong>create_temporary_system_function</strong>.
+        注册一个具有唯一名称的Python 自定义函数。 
+        可替换与该名称相同的已经存在的用户自定义函数。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:520]:
```suggestion
        若已有与该名称相同的用户自定义函数，则替换之。
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 13]
```diff
@@ -520,41 +516,41 @@ For more details about the different kinds of UDFs, please refer to [User Define
...
     </tr>
     <tr>
       <td>
         <strong>register_java_function(name, function_class_name)</strong>
       </td>
       <td>
-        Registers a Java user defined function under a unique name. 
-        Replaces already existing user-defined functions under this name.
-        It can be replaced by <strong>create_java_temporary_system_function</strong>.
+        注册一个具有唯一名称的 Java 自定义函数。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:532]: ditto

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 14]
```diff
@@ -563,47 +559,47 @@ Please refer to the [Dependency Management]({% link dev/python/table-api-users-g
...
+        <a href="{{ site.pythondocs_baseurl }}/api/python/pyflink.table.html#pyflink.table.TableEnvironment.set_python_requirements">链接</a>
       </td>
     </tr>
     <tr>
       <td>
         <strong>add_python_archive(archive_path, target_dir=None)</strong>
       </td>
       <td>
-        Adds a Python archive file. The file will be extracted to the working directory of Python UDF worker.
+        添加 Python 归档文件。该文件将被压缩到 Python UDF 程序的工作目录中。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:586]:
```suggestion
        添加 Python 归档文件。该文件将被解压到 Python UDF 程序的工作目录中。
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 15]
```diff
@@ -612,33 +608,33 @@ Please refer to the [Dependency Management]({% link dev/python/table-api-users-g
         <strong>get_config()</strong>
       </td>
       <td>
-        Returns the table config to define the runtime behavior of the Table API.
-        You can find all the available configuration options in <a href="{{ site.baseurl }}/zh/ops/config.html">Configuration</a> and
-        <a href="{{ site.baseurl }}/zh/dev/python/table-api-users-guide/python_config.html">Python Configuation</a>. <br> <br>
-        The following code is an example showing how to set the configuration options through this API:
+        通过 table config 来定义 Table API 的运行时行为。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:611]:
```suggestion
        返回 table config，可以通过 table config 来定义 Table API 的运行时行为。
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 16]
```diff
@@ -647,219 +643,219 @@ These APIs are used to access catalogs and modules. You can find more detailed i
...
       </td>
     </tr>
     <tr>
       <td>
         <strong>use_catalog(catalog_name)</strong>
       </td>
       <td>
-        Sets the current catalog to the given value.
-        It also sets the default database to the catalog's default one.
+        将当前目录设置为指定名称的 catalog。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:668]:
```suggestion
        将当前目录设置为所指定的 catalog。
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 17]
```diff
@@ -647,219 +643,219 @@ These APIs are used to access catalogs and modules. You can find more detailed i
...
+        它也可以将默认数据库设置为 catalog 的默认数据库。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:669]:
```suggestion
        它也将默认数据库设置为所指定 catalog 的默认数据库。
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 18]
```diff
@@ -647,219 +643,219 @@ These APIs are used to access catalogs and modules. You can find more detailed i
...
+        <a href="{{ site.pythondocs_baseurl }}/api/python/pyflink.table.html#pyflink.table.TableEnvironment.list_catalogs">链接</a>
       </td>
     </tr>
     <tr>
       <td>
         <strong>list_modules()</strong>
       </td>
       <td>
-        Gets the names of all modules registered in this environment.
+        获取在这个环境中注册的所有模块名称
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:749]:
```suggestion
        获取在这个环境中注册的所有模块名称。
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 19]
```diff
@@ -647,219 +643,219 @@ These APIs are used to access catalogs and modules. You can find more detailed i
...
+        <a href="{{ site.pythondocs_baseurl }}/api/python/pyflink.table.html#pyflink.table.TableEnvironment.list_views">链接</a>
       </td>
     </tr>
     <tr>
       <td>
         <strong>list_user_defined_functions()</strong>
       </td>
       <td>
-        Gets the names of all user defined functions registered in this environment.
+        获取在该环境中已注册的所有用户定义函数名称。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:795]:
```suggestion
        获取在该环境中已注册的所有用户自定义函数的名称。
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 20]
```diff
@@ -647,219 +643,219 @@ These APIs are used to access catalogs and modules. You can find more detailed i
...
 </table>
 
-Statebackend, Checkpoint and Restart Strategy
+Statebackend，Checkpoint 以及重启策略
 ---------------------------------------------
 
-Before Flink 1.10 you can configure the statebackend, checkpointing and restart strategy via the `StreamExecutionEnvironment`.
-And now you can configure them by setting key-value options in `TableConfig`, see [Fault Tolerance]({% link ops/config.zh.md %}#fault-tolerance), [State Backends]({% link ops/config.zh.md %}#checkpoints-and-state-backends) and [Checkpointing]({% link ops/config.zh.md %}#checkpointing) for more details.
+在 Flink 1.10 之前，你可以通过 `StreamExecutionEnvironment` 来配置 statebackend，checkpointing 以及重启策略。
+现在你可以通过在 `TableConfig` 中设置键值选项来配置它们，更多详情可查阅 [容错]({% link ops/config.zh.md %}#fault-tolerance)，[State Backends]({% link ops/config.zh.md %}#checkpoints-and-state-backends) 以及 [Checkpointing]({% link ops/config.zh.md %}#checkpointing)。
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:841]:
```suggestion
现在你可以通过在 `TableConfig` 中，通过设置键值选项来配置它们，更多详情可查阅 [容错]({% link ops/config.zh.md %}#fault-tolerance)，[State Backends]({% link ops/config.zh.md %}#checkpoints-and-state-backends) 以及 [Checkpointing]({% link ops/config.zh.md %}#checkpointing)。
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 21]
```diff
@@ -647,219 +643,219 @@ These APIs are used to access catalogs and modules. You can find more detailed i
...
 
-# set the checkpoint mode to EXACTLY_ONCE
+# 设置 checkpoint 模式为 EXACTLY_ONCE
 table_env.get_config().get_configuration().set_string("execution.checkpointing.mode", "EXACTLY_ONCE")
 table_env.get_config().get_configuration().set_string("execution.checkpointing.interval", "3min")
 
-# set the statebackend type to "rocksdb", other available options are "filesystem" and "jobmanager"
-# you can also set the full qualified Java class name of the StateBackendFactory to this option
+# 设置 statebackend 类型为 "rocksdb"，其他可用选项有 "filesystem" 和 "jobmanager"
+# 你也可以使用 StateBackendFactory 的完整类名来设置这个属性
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:855]:
```suggestion
# 你也可以将这个属性设置为 StateBackendFactory 的完整类名
```

File: docs/dev/python/table-api-users-guide/table_environment.zh.md [hunk 22]
```diff
@@ -647,219 +643,219 @@ These APIs are used to access catalogs and modules. You can find more detailed i
...
 table_env.get_config().get_configuration().set_string("restart-strategy.fixed-delay.delay", "30s")
 
-# set the checkpoint mode to EXACTLY_ONCE
+# 设置 checkpoint 模式为 EXACTLY_ONCE
 table_env.get_config().get_configuration().set_string("execution.checkpointing.mode", "EXACTLY_ONCE")
 table_env.get_config().get_configuration().set_string("execution.checkpointing.interval", "3min")
 
-# set the statebackend type to "rocksdb", other available options are "filesystem" and "jobmanager"
-# you can also set the full qualified Java class name of the StateBackendFactory to this option
+# 设置 statebackend 类型为 "rocksdb"，其他可用选项有 "filesystem" 和 "jobmanager"
```
Comment (dianfu) [inline docs/dev/python/table-api-users-guide/table_environment.zh.md:854]:
```suggestion
# 设置 statebackend 类型为 "rocksdb"，其他可选项有 "filesystem" 和 "jobmanager"
```

**Review by dianfu — COMMENTED** [review]
@wangxlong Thanks a lot for this work! Have left a few comments.

**Comment by wangxlong** [comment]
@dianfu  Updated.

**Review by dianfu — APPROVED** [review]

**Pull request closed without merging**